062-support-tiered-storage.md

```diff
@@ -0,0 +1,117 @@
+
+### RemoteLogMetadataManager (RLMM)
+
+For simplicity, the proposal doesn't add customization for RemoteLogMetadataManager. If tiered storage is enabled, the default implementation using `TopicBasedRemoteLogMetadataManager` will be used, and corresponding configuration will be appended. 
+
+Since the default RLMM will create an internal Kafka client to talk to the server, there might be some additional configuration needed, like the SSL related setting. Those config can be put in `.spec.kafka.config` directly using the `rlmm.config.` prefix. Example can be found below.
+
+### Example configuration
+
+The below config define a sample configuration for tiered storage setup, using a `custom` RSM type.
+```
+kafka:
+  tieredStorage:
+    type: custom
+    rsm.className: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+    rsm.classPath:  /opt/kafka/plugins/tiered-storage-s3/*
+    rsm.impl.prefix: rsm.config.
+
+  config:
+    ...
+    rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+    rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+```
+
+The configuration above will get translated to the below Kafka broker config:
+```
+remote.log.storage.system.enable: true
+remote.log.storage.manager.class.name: com.example.kafka.tiered.storage.s3.S3RemoteStorageManager
+remote.log.storage.manager.class.path: /opt/kafka/plugins/tiered-storage-s3/*
+remote.log.storage.manager.impl.prefix: rsm.config.
+# The RLMM config will be added by default
+remote.log.metadata.manager.impl.prefix: rlmm.config.
+remote.log.metadata.manager.class.name: org.apache.kafka.server.log.remote.metadata.storage.TopicBasedRemoteLogMetadataManager
+# The RLMM config below are passed down from kafka config spec.
+rlmm.config.remote.log.metadata.common.client.bootstrap.servers: broker-bootstrap.com:9091
+rlmm.config.remote.log.metadata.common.client.security.protocol: SSL
+```
+
+### Validation
+
+There are certain [limitations](https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Tiered+Storage+Early+Access+Release+Notes) for tiered storage enabled cluster.
+
+With Strimzi operator, we can validate the configuration before apply the change. Here are some validations we can apply:
+
+* Tiered storage should not be disabled at cluster level when there is any existing topic enabled tiered storage.
+
+* Tiered storage should not be enabled on a cluster with JBOD.
+
+* Tiered storage should not be applied for compact topic.
+
+### Docker image
+
+This proposal doesn't include built in plugin support in any Strimzi built Kafka image. Users need to specify the path via the config, and customize the docker image to include the library.
+
+An example to build such custom image is via a custom Dockerfile:
+
+```
+FROM quay.io/strimzi/kafka:latest-kafka-3.6.0
```

Comment:
Let's use a fixed version and not `latest`.

```suggestion
FROM quay.io/strimzi/kafka:0.40.0-kafka-3.6.0
```

Reply:
done